src/shogun/modelselection/NewGridSearch.cpp

```diff
@@ -0,0 +1,891 @@
+/*
+ * This software is distributed under BSD 3-clause license (see LICENSE file).
+ *
+ * Authors: Gil Hoben
+ */
+
+#include <shogun/distance/Distance.h>
+#include <shogun/evaluation/CrossValidation.h>
+#include <shogun/kernel/Kernel.h>
+#include <shogun/modelselection/NewGridSearch.h>
+#include <shogun/util/factory.h>
+
+using namespace shogun;
+
+const std::string ParameterNode::delimiter = "::";
+
+ParameterNode::ParameterNode(CSGObject* model)
+{
+	SG_REF(model)
+	m_parent = model;
+	SG_SDEBUG("%s refcount: %d\n", model->get_name(), m_parent->ref_count())
+	for (auto const& param : m_parent->get_params())
+	{
+		// check the param is neither of type CLabels or CFeatures
+		if ((std::type_index(param.second->get_value().type_info()) !=
```

Comment:
maybe have a central list of the types to exclude somewhere?
But thinking about this, the parameter nodes should only contain parameters that were registered as hyper-parameters in the classes that are passed. Then no filtering would be needed as CFeatures would never be marked as hyper-parameter for modelselection?

Reply:
True, eventually that is the goal!